modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/servlet/DDMFormTemplateContextProcessor.java

```diff
@@ -147,11 +147,17 @@ protected DDMFormField getDDMFormField(JSONObject jsonObject) {
 	protected DDMFormFieldOptions getDDMFormFieldOptions(JSONArray jsonArray) {
 		DDMFormFieldOptions ddmFormFieldOptions = new DDMFormFieldOptions();
 
+		String defaultLocale = null;
+
 		for (int i = 0; i < jsonArray.length(); i++) {
 			JSONObject jsonObject = jsonArray.getJSONObject(i);
 
 			String value = jsonObject.getString("value");
 
+			if ((defaultLocale == null) && jsonObject.has("defaultLocale")) {
+				defaultLocale = jsonObject.getString("defaultLocale");
+			}
```

Comment:
`jsonObject` is inside a FOR, so it can be all the items from the array. 
`defaultLocale` ccan change its value from one iteration to another. This is strange/unusual code :thinking: 

can you clarify or change the logic?